packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -884,6 +1256,221 @@ All @auth directives used on field definitions are performed when the field is r
         return rules.filter(rule => rule.allow === 'groups' && !Boolean(rule.groups));
     }
 
+    private extendTypeWithDirectives(ctx: TransformerContext, typeName: string, directives: DirectiveNode[]) {
+        let objectTypeExtension = blankObjectExtension(typeName);
+
+        objectTypeExtension = extensionWithDirectives(
+            objectTypeExtension,
+            directives
+        );
+
+        ctx.addObjectExtension(objectTypeExtension);
+    }
+
+    private addDirectivesToField(ctx: TransformerContext, typeName: string, fieldName: string, directives: DirectiveNode[]) {
+        const type = ctx.getType(typeName) as ObjectTypeDefinitionNode;
+
+        if (type) {
+            const field = type.fields.find((f) => f.name.value === fieldName);
+
+            if (field) {
+                const newFields = [
+                    ...type.fields.filter((f) => f.name.value !== field.name.value),
+                    extendFieldWithDirectives(field, directives)
+                ];
+
+                const newMutation = {
+                    ...type,
+                    fields: newFields
+                };
+
+                ctx.putType(newMutation);
+            }
+        }
+    }
+
+    private getDirectivesForRules(rules: AuthRule[], addDefaultIfNeeded: boolean = true): DirectiveNode[] {
+        if (!rules || rules.length === 0) {
+            return [];
+        }
+
+        const directives: DirectiveNode[] = new Array();
+
+        //
+        // We only add a directive if it is not the default auth or
+        // if it is the default one, but there are other rules for a
+        // different provider.
+        // For fields we don't add the default, since it would open up
+        // the access rights.
+        //
+
+        if ((this.configuredAuthProviders.default !== 'apiKey' &&
```

Comment:
I see duplicated code. Can we refactor this and parameterize over the expected auth provider string. You could define a helper closure inline.

Reply:
Refactored.